### PR TITLE
textboxwidget: enhanced CJK text wrapping

### PR DESCRIPTION
**Pull request by poire-z**
Followup from #2417 
> BTW, your CJK patches looks good to me, feel free to send a PR. I can help fix the unit tests. Today I learned that some of the characters in Japanese are not splitable :P

Implemented line breaking rules from :
https://en.wikipedia.org/wiki/Line_breaking_rules_in_East_Asian_languages
including the few japanese "Characters that can't be separated".
Note : I couldn't cut & paste the 4th character from this wikipedia page:
Korean `Characters that are not allowed at the end of a line :$([\{£`...
which appears as a \ in my paste ... so i let it out.

The existing unit tests may not fail, but they are not checking for the new breaking rules. So, you may extend the existing one or add new ones.

### CODE REVIEW

File: frontend/util.lua [hunk 1]
```diff
@@ -154,12 +154,55 @@ local non_splitable_space_tailers = ":;,.!?)]}$%=-+*/|<>»”"
 -- Same if a space has some specific other punctuation before it
 local non_splitable_space_leaders = "([{$=-+*/|<>«“"
 
+
+-- Similar rules exist for CJK text. Taken from :
+-- https://en.wikipedia.org/wiki/Line_breaking_rules_in_East_Asian_languages
+
+local cjk_non_splitable_tailers = table.concat( {
+-- Simplified Chinese
```
Comment (Frenzie) [inline frontend/util.lua]: don't forget to indent :-)

File: frontend/util.lua [hunk 2]
```diff
@@ -154,12 +154,55 @@ local non_splitable_space_tailers = ":;,.!?)]}$%=-+*/|<>»”"
...
+-- Traditional Chinese
+"!),.:;?]}¢·–—’\"•、。〆〞〕〉》」︰︱︲︳﹐﹑﹒﹓﹔﹕﹖﹘﹚﹜！），．：；？︶︸︺︼︾﹀﹂﹗］｜｝､",
+-- Japanese
+")]｝〕〉》」』】〙〗〟’\"｠»ヽヾーァィゥェォッャュョヮヵヶぁぃぅぇぉっゃゅょゎゕゖㇰㇱㇲㇳㇴㇵㇶㇷㇸㇹㇺㇻㇼㇽㇾㇿ々〻‐゠–〜?!‼⁇⁈⁉・、:;,。.",
+-- Korean
+"!%),.:;?]}¢°’\"†‡℃〆〈《「『〕！％），．：；？］｝",
+})
+
+local cjk_non_splitable_leaders = table.concat( {
+-- Simplified Chinese
```
Comment (Frenzie) [inline frontend/util.lua]: Indentation

**Comment by poire-z** [comment]
Done. (Thought these long lines would allow me to go unstylish, but ok :) - well, long for me when viewed with my non-utf8 editor, but they're shorter when viewed on github).

**Comment by houqp** [comment]
Thanks for the patch! You are a line breaking expert now ;)

**Pull request merged**